src/frontends/tensorflow_common/src/op/bincount.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "common_op_table.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/equal.hpp"
+#include "openvino/op/less.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/range.hpp"
+#include "openvino/op/reduce_sum.hpp"
+#include "openvino/op/select.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/unsqueeze.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+
+OutputVector translate_bincount_op(const NodeContext& node) {
+    default_op_checks(node, 3, {"Bincount"});
+    auto arr = node.get_input(0);
+    auto size = node.get_input(1);
+    auto weights = node.get_input(2);
+
+    std::vector<int32_t> size_scalar;
+    get_const_input(node, 1, &size_scalar);
+    TENSORFLOW_OP_VALIDATION(node, size_scalar.size() == 1, "size must be a scalar");
+    int32_t size_scalar_val = size_scalar[0];
+    TENSORFLOW_OP_VALIDATION(node, size_scalar_val > 0, "size must be non-negative.");
+
+    // weights: A Tensor. Must be one of the following types: int32, int64, float32, float64. is an int32, int64,
+    // float32, or float64 Tensor with the same shape as arr, or a length-0 Tensor, in which case it acts as all weights
+    // equal to 1.
+    auto weights_type = weights.get_element_type();
+    TENSORFLOW_OP_VALIDATION(node,
+                             weights_type == element::f32 || weights_type == element::f64 ||
+                                 weights_type == element::i32 || weights_type == element::i64,
+                             "Weights type is not supported: ",
+                             weights_type);
+    TENSORFLOW_OP_VALIDATION(node,
+                             arr.get_shape() == weights.get_shape() || shape_size(weights.get_shape()) == 0,
+                             "The shape of the weights must either match the first input or 0.");
+
+    if (shape_size(weights.get_shape()) == 0) {
+        // fill weighs with 1, the same shape with arr
+        weights = make_shared<v0::Constant>(weights_type, arr.get_shape(), std::vector<int>{1});
+    }
```

Comment:
please remove check for weights type for same reason as in above comments.

Also, let us check weights shape as `weights.get_partial_shape() == ov::Shape{0}` for which:
create scalar const node of type weights_type equal to 1 and broadcast it to `arr` shape. In order to compute arr shape, use `ShapeOf` operation and pass it to Broadcast for constant creation.
Otherwise we assume that no need to anything with weights.

Reply:
updated